server/src/lsp-server.spec.ts

```diff
@@ -184,6 +184,31 @@ describe('diagnostics', () => {
         await new Promise(resolve => setTimeout(resolve, 200));
         assert.equal(diagnosticsForThisTest.length, 2, JSON.stringify(diagnostics));
     }).timeout(10000);
+
+    it('code 80006 is ignored', async () => {
+        const doc = {
+            uri: uri('diagnosticsBar2.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+                function x() {
+                    return Promise.resolve(1).then(a => a + 1);
+                }
+                const i = 1;
+      `
+        }
+        server.didOpenTextDocument({
+            textDocument: doc
+        })
+
+        server.requestDiagnostics();
+        await server.requestDiagnostics();
+        await new Promise(resolve => setTimeout(resolve, 200));
+        const diagnosticsForThisFile = diagnostics.filter(d => d!.uri === doc.uri);
+        assert.equal(diagnosticsForThisFile.length, 1, JSON.stringify(diagnostics));
+        const fileDiagnostics = diagnosticsForThisFile[0]!.diagnostics;
+        assert.equal(fileDiagnostics.length, 0);
```

Comment:
This test passes but only because the sample TS code does not generate a diagnostic.

Reply:
But in reality, a diagnostic is emitted 
<img width="863" alt="Screenshot 2020-11-19 at 12 03 35" src="https://user-images.githubusercontent.com/145502/99658131-f7548e80-2a56-11eb-8bf2-eff2fa9571d9.png">

Reply:
Hrm, I'm not getting the same diagnostic error. I wonder what's up

But, switching the code to 
```
x() {
    return Promise.resolve(1).then(a => a + 1);
}
```

Does give me a diagnostic error:

```
  fileDiagnostics: [
    {
      range: [Object],
      message: "Cannot find name 'x'.",
      severity: 1,
      code: 2304,
      source: 'typescript',
      relatedInformation: undefined
    }
  ]
}

```

Which I can ignore by setting `ignoredDiagnosticCodes: [2304],` and that makes the tests pass

Reply:
I'll edit the test for this then.

Reply:
Maybe it has to do with the typescript version in use, or the target; I'm unclear as to which settings are used in the tests.

Reply:
Pushed the test change on [this fixup commit](https://github.com/pyrho/typescript-language-server/commit/c1666a2044951ea74b40739a274ae85dcaae932c)